### ISSUE TITLE
App crashes on launch

### ISSUE THREAD

**Issue by thomasRademaker** [realm/realm-swift#8089]
### How frequently does the bug occur?

Sometimes

### Description

a small percentage of users are reporting their app crashes on launch. I have never been able to reproduce.
They can never recover. Realm is in a bad state and the app crashes every time they try to open it.

### Stacktrace & log output

```shell
0  libsystem_kernel.dylib +0x7200  ___pthread_kill
1  libsystem_pthread.dylib +0x71a8 _pthread_kill
2  libsystem_c.dylib +0x20c88      _abort
3  Realm +0x2d6e84                 please_report_this_issue_in_github_realm_realm_core_v_12_13_0
4  Realm +0x2d716c                 realm::util::terminate_internal(std::__1::basic_stringstream<char, std::__1::char_traits<char>, std::__1::allocator<char> >&)
5  Realm +0x2d6fcc                 realm::util::terminate(char const*, char const*, long, std::initializer_list<realm::util::Printable>&&)
6  Realm +0x17b684                 realm::Cluster::init(realm::MemRef)
7  Realm +0x18c09c                 realm::ClusterTree::create_root_from_parent(realm::ArrayParent*, unsigned long)
8  Realm +0x18c3c8                 realm::ClusterTree::init_from_parent()
9  Realm +0x285360                 realm::Table::init(unsigned long, realm::ArrayParent*, unsigned long, bool, bool)
10 Realm +0x1a9244                 realm::Group::create_table_accessor(unsigned long)
11 Realm +0x1a8ec0                 realm::Group::do_get_table(unsigned long)
12 Realm +0x2b0604                 realm::Transaction::check_consistency()
13 Realm +0x18faa4                 realm::DB::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool, realm::DBOptions const&)
14 Realm +0x1969a8                 realm::DB::create(std::__1::unique_ptr<realm::Replication, std::__1::default_delete<realm::Replication> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, realm::DBOptions const&)
15 Realm +0x3bfcec                 realm::_impl::RealmCoordinator::open_db()
16 Realm +0x3c0a00                 realm::_impl::RealmCoordinator::do_get_realm(realm::RealmConfig, std::__1::shared_ptr<realm::Realm>&, std::__1::optional<realm::VersionID>, realm::util::CheckedUniqueLock&)
17 Realm +0x3c086c                 realm::_impl::RealmCoordinator::get_realm(realm::RealmConfig, std::__1::optional<realm::VersionID>)
18 Realm +0x44e424                 realm::Realm::get_shared_realm(realm::RealmConfig)
19 Realm +0x126874                 +[RLMRealm realmWithConfiguration:queue:error:] (RLMRealm.mm:532:25)
20 RealmSwift +0x6f174             @nonobjc RLMRealm.__allocating_init(configuration:queue:) (<compiler-generated>)
21 RealmSwift +0x6f684             Realm.init(configuration:queue:) (Realm.swift:95:28)
22 AppNamePod +0xb415c           static DataManager.setup(schemaVersion:inMemoryIdentifier:deleteRealmIfMigrationNeeded:appGroup:) (DataManager.swift:42:21)
23 AppNamePod +0x891d4           setup(environment:schemaVersion:inMemoryIdentifier:useAsCache:appGroup:) (AppNamePod.swift:33:17)
24 AppName +0x45970               specialized AppDelegate.setupAppNamePod(_:launchOptions:) (AppDelegate.swift:130:9)
25 AppName +0x44aa0               AppDelegate.setupAppNamePod(_:launchOptions:) (<compiler-generated>)
26 AppName +0x44dbc               AppDelegate.application(_:didFinishLaunchingWithOptions:) (AppDelegate.swift:44:9)
27 UIKitCore +0x35a884             -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:]
28 UIKitCore +0x359fa8             -[UIApplication _callInitializationDelegatesWithActions:forCanvas:payload:fromOriginatingProcess:]
29 UIKitCore +0x358f84             -[UIApplication _runWithMainScene:transitionContext:completion:]
30 UIKitCore +0x358bd0             -[_UISceneLifecycleMultiplexer completeApplicationLaunchWithFBSScene:transitionContext:]
31 UIKitCore +0x9e5fc              __UIScenePerformActionsWithLifecycleActionMask
32 UIKitCore +0x3e0914             ___101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke
33 UIKitCore +0x290fa0             -[_UISceneLifecycleMultiplexer _performBlock:withApplicationOfDeactivationReasons:fromReasons:]
34 UIKitCore +0x290dc8             -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]
35 UIKitCore +0x290978             -[_UISceneLifecycleMultiplexer uiScene:transitionedFromState:withTransitionContext:]
36 UIKitCore +0x290844             ___186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke
37 UIKitCore +0x972fa4             +[BSAnimationSettings(UIKit) tryAnimatingWithSettings:fromCurrentState:actions:completion:]
38 UIKitCore +0xa0bf94             __UISceneSettingsDiffActionPerformChangesWithTransitionContextAndCompletion
39 UIKitCore +0x13c954             -[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]
40 UIKitCore +0x5af7a4             ___64-[UIScene scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke.214
41 UIKitCore +0x20f0b4             -[UIScene _emitSceneSettingsUpdateResponseForCompletion:afterSceneUpdateWork:]
42 UIKitCore +0x20ef24             -[UIScene scene:didUpdateWithDiff:transitionContext:completion:]
43 UIKitCore +0x20e478             -[UIApplication workspace:didCreateScene:withTransitionContext:completion:]
44 UIKitCore +0x20e204             -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:]
45 FrontBoardServices +0x34fc      -[FBSScene _callOutQueue_agent_didCreateWithTransitionContext:completion:]
46 FrontBoardServices +0x42518     ___92-[FBSWorkspaceScenesClient createSceneWithIdentity:parameters:transitionContext:completion:]_block_invoke.78
47 FrontBoardServices +0x7290      -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:]
48 FrontBoardServices +0x42150     ___92-[FBSWorkspaceScenesClient createSceneWithIdentity:parameters:transitionContext:completion:]_block_invoke
49 libdispatch.dylib +0x3fd8       __dispatch_client_callout
50 libdispatch.dylib +0x7a58       __dispatch_block_invoke_direct
51 FrontBoardServices +0x113ac     __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__
52 FrontBoardServices +0x10f48     -[FBSSerialQueue _targetQueue_performNextIfPossible]
53 FrontBoardServices +0x13728     -[FBSSerialQueue _performNextFromRunLoopSource]
54 CoreFoundation +0xd5f50         ___CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
55 CoreFoundation +0xe2328         ___CFRunLoopDoSource0
56 CoreFoundation +0x6626c         ___CFRunLoopDoSources0
57 CoreFoundation +0x7bba4         ___CFRunLoopRun
58 CoreFoundation +0x80ed0         _CFRunLoopRunSpecific
59 GraphicsServices +0x1364        _GSEventRunModal
60 UIKitCore +0x3a23cc             -[UIApplication _run]
61 UIKitCore +0x3a2030             _UIApplicationMain
62 AppName +0x46f0                main (AppDelegate.swift:25:13)
63 dyld +0x1595c                   start
```

### Can you reproduce the bug?

Not yet

### Version

10.32.2

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

iOS 15.6.1 - 16.2

### Build environment

Xcode version: 14.2
Dependency manager and version: cocoapods 1.11.3
I've seen a similar issue with SPM

**Comment by dianaafanador3** [comment]
HI @thomasRademaker, some corruption issues has been fixed here https://github.com/realm/realm-core/pull/5993 recently which can be related to this. A new version will be released shortly with the fix and we can check after the release to see if we are still hitting those assertions.
This fix will not fix the corrupted files, but will prevent future corruption cases.

**Comment by thomasRademaker** [comment]
thanks for the feedback @dianaafanador3 
> This fix will not fix the corrupted files, but will prevent future corruption cases.

Is the only way to "fix" the corrupted files is to delete the Realm file?

**Comment by dianaafanador3** [comment]
@thomasRademaker If you have contact with the user and get a hold of the file, we could help to fix the file. If not, the best solution will be to reinstall the app and create a new realm file. 
My suspicion is that it is a corruption file issue, but if you can reproduce it and send us a sample code with the repro case, we can take a closer look to it.

**Comment by thomasRademaker** [comment]
@dianaafanador3 Is this live in a release https://github.com/realm/realm-core/pull/5993 ? 

we are using Realm as a cache, so no important user data will be lost if we delete a realm from disk.
I will create a new realm at a new file path and delete the realm at the old file path.

I think you can close this issue. Thank you for your help.

**Comment by rodrigobertolottizinsp** [comment]
@dianaafanador3 can you confirm if this was actually released in any recent versions? 
thank you!

**Comment by rodrigobertolottizinsp** [comment]
@dianaafanador3 can you confirm if this was actually released in any recent versions? 
thank you!

**Comment by tgoyne** [comment]
That was released in 10.33.0.

**Comment by thomasRademaker** [comment]
Thank you. We have released a new build with this version of Realm and the issue is gone. This issue can be closed. Thank you
